### PR TITLE
feat(BDropdown): add keyboard navigation

### PR DESCRIPTION
**Pull request by xvaara**
# Describe the PR

A clear and concise description of what the pull request does.

## Small replication

A small replication or video walkthrough can help demonstrate the changes made. This is optional, but can help observe the intended changes. A mentioned issue that contains a replication also works.

## PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix :bug: - `fix(...)`
- [ ] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [ ] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [ ] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

### CODE REVIEW

File: packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue [hunk 1]
```diff
@@ -171,15 +171,34 @@ const rootBoundary = computed<RootBoundary | undefined>(() =>
...
-  () => {
-    modelValue.value = !modelValueBoolean
-  },
-  {target: splitButton}
-)
-
 const referencePlacement = toRef(() => (!splitBoolean.value ? splitButton.value : button.value))
+
+onKeyStroke('Escape', () => (modelValue.value = !modelValueBoolean), {target: referencePlacement})
+onKeyStroke('Escape', () => (modelValue.value = !modelValueBoolean), {target: floating})
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue:177]:
```suggestion
onKeyStroke('Escape', () => {modelValue.value = !modelValueBoolean}, {target: referencePlacement})
onKeyStroke('Escape', () => {modelValue.value = !modelValueBoolean}, {target: floating})
```
Whether or not it matters in this context, 

When you do 
```ts
let a = 1
const b = () => (a = 2)
```
The return type of b is not `void`, the return type is `number`. So, I prefer to never use that syntax due to unexpected complications

File: packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue [hunk 2]
```diff
@@ -171,15 +171,34 @@ const rootBoundary = computed<RootBoundary | undefined>(() =>
...
+
+function keynav(e: Event, v: number) {
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue:179]:
```suggestion
const keynav = (e: Event, v: number) => {
```

File: packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue [hunk 3]
```diff
@@ -171,15 +171,34 @@ const rootBoundary = computed<RootBoundary | undefined>(() =>
...
+  e.preventDefault()
+  if (!modelValueBoolean.value) {
+    open()
+    nextTick(() => keynav(e, v))
+    return
+  }
+  const list = floating.value?.querySelectorAll('.dropdown-item:not(.disabled):not(:disabled)')
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue:186]: Have you made sure that disabled items are skipped properly?

```html
<b-dropdown>
    <b-dropdown-item>First Action</b-dropdown-item>
    <b-dropdown-item disabled>Disabled action</b-dropdown-item>
    <b-dropdown-item>Third Action</b-dropdown-item>
  </b-dropdown>
```
Does this function as expected?

**Review by VividLemon — CHANGES_REQUESTED** [review]

File: packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue [hunk 4]
```diff
@@ -171,15 +171,34 @@ const rootBoundary = computed<RootBoundary | undefined>(() =>
   props.boundary === 'document' || props.boundary === 'viewport' ? props.boundary : undefined
 )
 
-onKeyStroke(
-  'Escape',
-  () => {
-    modelValue.value = !modelValueBoolean
-  },
-  {target: splitButton}
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue:179]: Make sure that you're targeting the right items. I saw that this had changed, and I want to make sure that it's correct. I think referencePlacement is correct for this, but just making sure.

File: packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue (hunk 1 above)
Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue:177]: I think that is some eslint or something completed it like that. Will change.

File: packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue (hunk 3 above)
Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue:186]: It should. the :not(:disabled) should skip those.

File: packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue (hunk 4 above)
Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue:179]: Yes, it wasn't working correctly before.

**Review by VividLemon — APPROVED** [review]

**Pull request merged**